lib/kafka_ex/protocol/common.ex

```diff
@@ -0,0 +1,16 @@
+defmodule KafkaEx.Protocol.Common do
+  @moduledoc """
+  A collection of common request generation and response parsing functions for the
+  Kafka wire protocol.
+  """
+
+  @doc """
+  Generate the wire representation for a list of topics.
+  """
+  def topic_data([]), do: ""
+
+  def topic_data([topic|topics]) do
```

Comment:
Huh.  I didn't know.  It still says tail-recursive tends to be faster on x86 hardware _shrugs_.